src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3069 @@
+        BG_ClearCharDataRange(GF_BG_LYR_MAIN_0, 0x20, 0x0000, HEAP_ID_REGISTER_HALL_OF_FAME);
+    }
+
+    {
+        BgTemplate bgTemplate = sBgTemplate_IndivMons_Main1;
+        InitBgFromTemplate(data->bgConfig, GF_BG_LYR_MAIN_1, &bgTemplate, GF_BG_TYPE_TEXT);
+    }
+
+    {
+        BgTemplate bgTemplate = sBgTemplate_IndivMons_Main2;
+        InitBgFromTemplate(data->bgConfig, GF_BG_LYR_MAIN_2, &bgTemplate, GF_BG_TYPE_TEXT);
+    }
+
+    {
+        BgTemplate bgTemplate = sBgTemplate_IndivMons_Main3;
+        InitBgFromTemplate(data->bgConfig, GF_BG_LYR_MAIN_3, &bgTemplate, GF_BG_TYPE_TEXT);
+    }
+}
+
+static void RegisterHallOfFame_IndivMons_DeleteBgTilemapBuffers(RegisterHallOfFameData *data) {
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_3);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_2);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_1);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_0);
+}
+
+static void RegisterHallOfFames_IndivMons_LoadBgGfx(RegisterHallOfFameData *data) {
+    GfGfxLoader_LoadScrnDataFromOpenNarc(data->narcA094, 0, data->bgConfig, GF_BG_LYR_MAIN_1, 0, 0, TRUE, HEAP_ID_REGISTER_HALL_OF_FAME);
+    GfGfxLoader_LoadScrnDataFromOpenNarc(data->narcA094, 1, data->bgConfig, GF_BG_LYR_MAIN_2, 0, 0, TRUE, HEAP_ID_REGISTER_HALL_OF_FAME);
+    GfGfxLoader_LoadScrnDataFromOpenNarc(data->narcA094, 2, data->bgConfig, GF_BG_LYR_MAIN_3, 0, 0, TRUE, HEAP_ID_REGISTER_HALL_OF_FAME);
+    GfGfxLoader_LoadCharDataFromOpenNarc(data->narcA094, 3, data->bgConfig, GF_BG_LYR_MAIN_1, 0, 0, TRUE, HEAP_ID_REGISTER_HALL_OF_FAME);
+    GfGfxLoader_GXLoadPalFromOpenNarc(data->narcA094, 4, GF_PAL_LOCATION_MAIN_BG, (enum GFPalSlotOffset)0, 0x20, HEAP_ID_REGISTER_HALL_OF_FAME);
+}
+
+static void RegisterHallOfFame_IndivMonsScene_InitWindows(RegisterHallOfFameData *data) {
+    for (int i = 0; i < 7u; ++i) {
+        AddWindow(data->bgConfig, &data->windows[i], &sIndivMonsSceneWindowTemplates[i]);
+    }
+
+    FillWindowPixelBuffer(&data->windows[0], 0);
+    ReadMsgDataIntoString(data->msgData, msg_0180_00000, data->strbuf1);  // The Hall of Fame!
+    AddTextPrinterParameterizedWithColor(&data->windows[0], 0, data->strbuf1, 128 - FontID_String_GetWidth(0, data->strbuf1, 0) / 2, 0, TEXT_SPEED_NOTRANSFER, MAKE_TEXT_COLOR(15, 2, 0), NULL);
+    CopyWindowPixelsToVram_TextMode(&data->windows[0]);
+}
+
+static void RegisterHallOfFame_IndivMonsScene_RemoveWindows(RegisterHallOfFameData *data) {
+    for (int i = 0; i < 7u; ++i) {
+        RemoveWindow(&data->windows[i]);
+    }
+}
+
+static void RegisterHallOfFame_IndivMonsScene_PrintMonDetailsOnWindow(RegisterHallOfFameData *data, u16 monIdx, int windowIdx) {
+    Window *windows = &data->windows[windowIdx];
+    RegisterHofMon *hofMon = &data->mons[monIdx];
+    Pokemon *mon = hofMon->mon;
+    BoxPokemon *boxmon = Mon_GetBoxMon(mon);
+
+    FillWindowPixelBuffer(&windows[0], 0);
+    ReadMsgDataIntoString(data->msgData, msg_0180_00004, data->strbuf1);
```

Comment:
Add comment with message contents

Reply:
etc for function

Reply:
the msgdata headers are already commented